src/api/libcellml/entity.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.Some license of other
+*/
+
+#ifndef LIBCELLML_LIBCELLML_ENTITY_H_
+#define LIBCELLML_LIBCELLML_ENTITY_H_
+
+#include "libcellml/enumerations.h"
+
+namespace libcellml {
+
+/**
+ * @brief The Entity class.
+ * Base class for all serialisable libCellML classes.
+ */
+class Entity
+{
+protected:
+    /**
+     * @brief Pure virtual serialise method.
+     * Pure virtual serialise method for serialising a CellML object to a std::string.
+     * @param format The format to serialise the object to.
+     * @return std::string representation of the object.
+     */
+    virtual std::string serialise(libcellml::CELLML_FORMATS format) const = 0;
```

Comment:
In the case of the `format` parameter it is in this case an enumeration and therefore it is really weird to pass it as a `const` reference in my opinion.  But otherwise yes in parameters should be passed by const reference if appropriate.

As for prefixing the parameter the Google style guide doesn't explicitly mention parameter naming, but in all the examples offered none have a parameter prefixed with anything.  I am against adding an unnecessary prefix and would not like to see as part of the libcellml coding standard.

For class members I am ambivalent about the prefix (the Google style guide here says class members should be post-fixed with an underscore) either pre-fixing with `_` or `m` is fine for me.  But we should decide and add it to the libcellml standard.